galaxy_ng/app/pipelines.py

```diff
@@ -0,0 +1,47 @@
+from django.conf import settings
+from django.contrib.auth.models import Group
+
+
+def user_role(response, details, user=None, *args, **kwargs):
+    if not user:
+        return
+
+    roles = response.get(settings.KEYCLOAK_ROLE_TOKEN_CLAIM, [])
+    if not roles:
+        return
+
+    if not isinstance(roles, list):
+        return
+
+    is_admin = False
+    if settings.KEYCLOAK_ADMIN_ROLE in roles:
+        is_admin = True
+
+    user.is_staff = is_admin
+    user.is_admin = is_admin
+    user.is_superuser = is_admin
+    user.save()
+
+
+def user_group(response, details, user=None, *args, **kwargs):
```

Comment:
Groups don't get updated with DRF token authentication, which creates a security risk if someone's access to a group is removed and they only ever authenticate using tokens.

```
curl \
  -X GET \
  -H 'Authorization: Token dc52720eaf83f0125a31722094060c127ad0b8ab' -H 'Content-Type: application/json' \
  http://localhost:5001/api/automation-hub/_ui/v1/me/
```

Reply:
Capturing results of offline conversation here to resolve the above issue:

- Enhance code with token expiration for Social Auth user which will require regular logins to pick up group associations.
- Management commands for creating/deleting “service accounts” (i.e. Django only user)
- Management command for obtaining a token for a "service account"